astropy/modeling/core.py

```diff
@@ -1224,8 +1365,115 @@ def _param_sets(self, raw=False):
             psets[:] = values
             return psets
 
+        # TODO: Returning an array from this method may be entirely pointless
+        # for internal use--perhaps only the external param_sets method should
+        # return an array (and just for backwards compat--I would prefer to
+        # maybe deprecate that method)
+
         return np.array(values)
 
+    def _prepare_input_units(self, inputs):
+        for input_ in inputs:
```

Comment:
Mostly going through to try to understand it, but since `any` short-circuits, this can just be:
```
self._using_quantities |= any(isinstance(input_, Quantity) for input_ in inputs)
```
(which I think is more readable).